src/query/storage/m3/dynamic_cluster.go

```diff
@@ -177,6 +180,11 @@ func (d *dynamicCluster) updateNamespaces(
 	d.updateNamespacesByEtcdClusterWithLock(etcdClusterID, clusterCfg, newNamespaces)
 	d.updateClusterNamespacesWithLock()
 	d.Unlock()
+
+	if err := d.clusterNamespacesWatcher.Update(d.ClusterNamespaces()); err != nil {
+		d.logger.Error("failed to update cluster namespaces watcher. downstream components relying on namespace "+
```

Comment:
nit; do these need to be this verbose? e.g. compared to `errClusterNamespacesWatcherNotSet` above

Reply:
Nah probably not. Will remove the detail here.